import_shims/warn.py

```diff
@@ -37,10 +38,30 @@ def __str__(self):
         ).format(self=self)
 
 
+class DeprecatedEdxPlatformImportError(Exception):
+    """
+    Error: An edx-platform module is being imported from an unsupported location.
+
+    See `DeprecatedEdxPlatformImportWarning` above for context.
```

Comment:
NIT: Isn't this warning not going to get printed because you raise before printing the warning below?

Reply:
Sorry, "above" in the source file.

I'll leave this as-is given that it's getting deleted next week (hopefully), but the ambiguity is noted :)